deeppavlov/dataset_readers/md_yaml_dialogs_reader.py

```diff
@@ -0,0 +1,331 @@
+
+                curr_story_utters.append(system_action)
+
+        stories_parsed.pop(None)
+
+        tmp_f = tempfile.NamedTemporaryFile(delete=False, mode='w', encoding="utf-8")
+        for story_id, story in stories_parsed.items():
+            for replics in story:
+                print(json.dumps(replics), file=tmp_f)
+            print(file=tmp_f)
+        tmp_f.close()
+        gobot_formatted_stories = DSTC2DatasetReader._read_from_file(tmp_f.name, dialogs=dialogs)
+        os.remove(tmp_f.name)
+        return gobot_formatted_stories
+
+    @classmethod
+    def _choose_slots_for_whom_exists_text(cls, intent2slots2text, slots_actual_values, user_action):
+        slots_used_values = slots_actual_values
+        slots_to_exclude = []
+        if not intent2slots2text[user_action].get(slots_actual_values):
+            slots_lazy_key = set(e[0] for e in slots_actual_values)
+            fake_keys = []
+            for known_key in intent2slots2text[user_action].keys():
+                if slots_lazy_key.issubset(set(e[0] for e in known_key)):
+                    fake_keys.append(known_key)
+                    break
+            fake_key = None
+            if fake_keys:
+                fake_key = sorted(fake_keys,
+                                  key=lambda elem: (len(set(slots_actual_values) ^ set(elem)),
+                                                    len([e for e in elem if e[0] not in slots_lazy_key])))[0]
+            slots_to_exclude = [e[0] for e in fake_key if e[0] not in slots_lazy_key]
+            slots_used_values = fake_key
+        return slots_to_exclude, slots_used_values
+
+    @classmethod
+    def _clarify_slots_values(cls, slot_name2text2value, slots_dstc2formatted):
+        slots_key = []
+        for slot_name, slot_value in slots_dstc2formatted:
+            if slot_value in slot_name2text2value[slot_name].keys():
+                slot_actual_value = slot_name2text2value[slot_name][slot_value]
+            else:
+                slot_actual_value = slot_value
+            slots_key.append((slot_name, slot_actual_value))
+        slots_key = tuple(sorted(slots_key))
+        return slots_key
+
+    @classmethod
+    def _parse_user_intent(cls, line):
+        intent = line.strip('*').strip()
+        if '{' not in intent:
+            intent = intent + "{}"  # the prototypical intent is "intent_name{slot1: value1, slotN: valueN}"
+        user_action, slots_info = intent.split('{', 1)
+        slots_info = json.loads('{' + slots_info)
+        slots_dstc2formatted = [[slot_name, slot_value] for slot_name, slot_value in slots_info.items()]
+        return user_action, slots_dstc2formatted
+
+    @classmethod
+    def _user_text(cls, intent2slots2text: dict, user_action: str, slots_li: dict = {}):
+        return intent2slots2text[user_action][slots_li][0]
+
+    @classmethod
+    def _system_action2text(cls, domain_knowledge: DomainKnowledge, system_action: str):
```

Comment:
method name should be the same as for _user_text

Reply:
to keep things more straightforward I'll rename _user_text to _user_action2text.